mathics/builtin/arithmetic.py

```diff
@@ -634,11 +652,12 @@ def apply(self, expr, evaluation):
             if numeric_val and hasattr(numeric_val, "is_approx_zero"):
                 result = numeric_val.is_approx_zero
             elif (
-                Expression("NumericQ", numeric_val).evaluate(evaluation) == SymbolFalse
+                Expression(SymbolNumericQ, numeric_val).evaluate(evaluation)
+                is SymbolFalse
             ):
                 return (
                     SymbolTrue
-                    if Expression("Simplify", expr).evaluate(evaluation) == Integer0
+                    if Expression(SymbolSimplify, expr).evaluate(evaluation) == Integer0
```

Comment:
Maybe `is` instead of `==`?

Reply:
Nop, because if for some reason Expression(SymbolSimplify, expr) evaluates() to something like
``
Integer(3)-Integer(1)-Integer(2)
``
the result is a new instance of Integer(0) that does not necessarily is the same that the system `Integer0`...